docs/source/articles/sequential-order-numbering.html.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Sequential Order Numbering
+created_at: 2020/09/24
+excerpt: "In Workarea, orders (and other persisted data) are identified in a different manner from relational databases in order to avoid collisions. This guide explains how Workarea identifies orders, and how you might want to customize this for vanity purposes."
+---
+
+# Sequential Order Numbering
+
+In relational database systems, identifiers are typically numbers that are automatically incremented. Many ecommerce systems, both in the past and still running today, are built around relational databases in order to persist their data. As a result, order identifiers tend to have a "numerical" look to them, so much so that the colloqualism "order number" has been adopted throughout the industry. In Workarea, orders (and other persisted data) are identified in a different manner to avoid collisions when the backing database store is distributed amongst many machines. This guide explains how Workarea identifies orders, and how you might want to customize this for vanity purposes.
+
+## How Workarea Identifies Orders
+
+When an order is created or instantiated, it is assigned a randomized 5-digit uppercase hexadecimal value as its ID. Workarea does this so order numbers are legible to human eyes, and will play nicely on a distributed database system such as MongoDB. This ID is used throughout the system to link orders to their Payment and Shipping information, as well as in the admin for display purposes. Although this order ID is "randomized", `SecureRandom` generation uses the current system time to determine a seed value for its algorithm, so when compared, they are actually sequential already:
+
+```ruby
+def test_sequential_order_numbering
+  orders = [create_order, create_order]
+
+  assert(orders.second < orders.first) # => true
+end
+```
+
+However, Workarea's order IDs definitely don't look "numerical", and may be confusing to read for those used to relational systems. For this reason, you may want to customize how admins view orders on their site.
```

Comment:
```suggestion
However, Workarea's order IDs definitely don't look numerical, and may not be what people are used to. For this reason, merchants may insist on customizing how admins view orders on their site.
```

1) Let's not create the thought they might be confusing and 2) let's imply they are pushing back (which they should be). 😄

Reply:
I like the second part, but I'm not sure we're "creating" the thought here for merchants if they're asking for this in the first place. I will reword it, however, to drive home the point that a _merchant_ is the one who asks for this, not a developer.

Reply:
What I mean is they might not have been thinking about that and now by mentioning it they may go "hmm, _are_ these confusing?"